File: Programming/Lv.1/약수의 개수와 덧셈/solution.py
# https://school.programmers.co.kr/learn/courses/30/lessons/77884?language=python3
import math

def solution(left, right):
    answer = 0
    for n in range(left, right + 1):
        answer += -n if math.sqrt(n).is_integer() else n
        
    return answer

File: Programming/Lv.1/약수의 개수와 덧셈/solution.js
// https://school.programmers.co.kr/learn/courses/30/lessons/77884
function solution(left, right) {
  var answer = 0;

  for (let i = left; i <= right; i += 1) {
    const divisors = [];
    for (let j = 1; j <= i / 2; j += 1) {
      if (i % j === 0) {
        divisors.push(j);
      }
    }
    divisors.push(i);
    answer += divisors.length % 2 === 0 ? i : -i;
  }

  return answer;
}
